influxdb/client.py

```diff
@@ -203,17 +205,13 @@ def request(self, url, method='GET', params=None, data=None,
         """
         url = "{0}/{1}".format(self._baseurl, url)
 
+        if headers is None:
+            headers = self._headers
+
         if params is None:
             params = {}
 
-        auth = {
-            'u': self._username,
-            'p': self._password
-        }
-
-        params.update(auth)
-
-        if data is not None and not isinstance(data, str):
+        if isinstance(data, dict) or isinstance(data, list):
```

Comment:
AFAIK, `isinstance` supports a tuple as the second argument, so this line can be written as `if isinstance(data, (dict, list)):`.